pkg/database/user_report_test.go

```diff
@@ -124,21 +124,49 @@ func TestDeleteUserReport(t *testing.T) {
 	t.Parallel()
 
 	db, _ := testDatabaseInstance.NewDatabase(t, nil)
+	realm, err := db.FindRealm(1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	realm.AddUserReportToAllowedTestTypes()
+	if err := db.SaveRealm(realm, SystemTest); err != nil {
+		t.Fatal(err)
+	}
 
 	phoneNumber := "+8128675309"
-	userReport, err := db.NewUserReport(phoneNumber, generateNonce(t), true)
-	if err != nil {
-		t.Fatalf("error creating user report: %v", err)
+	verCode := VerificationCode{
+		RealmID:       realm.ID,
+		Code:          "12345678",
+		LongCode:      "12345678",
+		TestType:      "user-report",
+		PhoneNumber:   phoneNumber,
+		ExpiresAt:     time.Now().Add(time.Hour),
+		LongExpiresAt: time.Now().Add(time.Hour),
 	}
-	if err := db.db.Save(userReport).Error; err != nil {
-		t.Fatalf("error writing user report: %v", err)
+
+	if err := realm.SaveVerificationCode(db, &verCode); err != nil {
+		t.Fatalf("unable to save code and user report: %v", err)
 	}
 
-	if err := db.DeleteUserReport(phoneNumber); err != nil {
+	var got *UserReport
+	if err := db.db.Transaction(
```

Comment:
Why use a tx here?

Reply:
because the FindUserReport requires a tx?